core/src/consensus/tendermint/engine.rs

```diff
@@ -472,6 +472,12 @@ fn give_additional_rewards<F: FnMut(&Address, u64) -> Result<(), Error>>(
     let sorted_validators = missed_signatures
         .into_iter()
         .map(|(address, (proposed, missed))| (address, Ratio::new(missed, proposed)))
+        // When one sees the Ratio crate's Order trait implementation, he/she can easily realize the
+        // comparing routine is erroneous. It inversely compares denominators when the numerators are same.
+        // That's problematic when it compares such as Ratio{numer: 0, denom:2} and Ratio{numer: 0, denom: 3}.
+        // But the Ratio::new() calls Ratio::reduce() inside it so that the numerator and the denominator are
+        // divided equally by gcd. Therefore the comparison between above two instances will never happen and only
```

Comment:
// divided equally by the gcd. Therefore the comparison between above two instances would never happen and only

Reply:
Accepted